### DISCUSSION TITLE
Using MFX Modal dialogs

### DISCUSSION THREAD

**Discussion by infinite-dev22** [palexdev/MaterialFX#311] [Q&A]
Hello @palexdev, I need guidance on how to use `MFXGenericDialogBuilder`, `MFXGenericDialog` and `MFXStageDialog` to create a Modal Dialog with Input controls from which am to pick the data inserted and store it. I have tried `MFXGenericDialogBuilder.build().setContent(dialogContentNode())` where `dialogContentNode()` is a function returning a Node and `MFXGenericDialogBuilder.build().setContent(dialogNode)` where `dialogNode` is a `HBox` in an `FXML` file and all return an empty Modal Dialog.

![Screenshot from 2023-04-21 18-08-02](https://user-images.githubusercontent.com/77667007/233671307-123d0c9f-2d54-4771-bc81-575c6fedc0f6.png)

![Screenshot from 2023-04-21 18-07-52](https://user-images.githubusercontent.com/77667007/233671367-d842cc6d-af7a-49fe-93f1-0510a200cef2.png)

**Comment by faustologitrack** [comment]
This works for me

```
public void showCustomModal(Node contentNode, String titulo) {
    MFXGenericDialog dialogContent = MFXGenericDialogBuilder.build()
            .setHeaderText(titulo)
            .makeScrollable(true)
            .get();

    MFXStageDialog stageDialog = MFXStageDialogBuilder.build(new MFXStageDialog(dialogContent))
            .initOwner(this.logintButton.getScene().getWindow())
            .initModality(Modality.APPLICATION_MODAL)
            .setDraggable(true)
            .setOwnerNode(this.loginStackPane)
            .setScrimPriority(ScrimPriority.WINDOW)
            .setScrimOwner(true)
            .get();

    dialogContent.addActions(
            Map.entry(new MFXButton("Confirm"), event -> {
                // TODO
            }),
            Map.entry(new MFXButton("Cancel"), event -> {
                stageDialog.close();
            }));

    // add content after stageDialog was built
    dialogContent.setContent(contentNode);

    stageDialog.showDialog();
}
```
